config/main.py

```diff
@@ -1410,38 +1433,109 @@ def qos(ctx):
     pass
 
 @qos.command('clear')
-def clear():
+@click.option(
+    '-n', '--namespace', default=None, type=str, help="Network namespace"
+)
+def clear(namespace):
     """Clear QoS configuration"""
     log_info("'qos clear' executing...")
-    _clear_qos()
+    _clear_qos(namespace)
 
 @qos.command('reload')
-def reload():
+@click.option(
+    '-n', '--namespace', default=None, type=str, help="Network namespace"
+)
+def reload(namespace):
     """Reload QoS configuration"""
     log_info("'qos reload' executing...")
-    _clear_qos()
+    _clear_qos(namespace)
     platform = sonic_device_util.get_platform()
     hwsku = sonic_device_util.get_hwsku()
-    buffer_template_file = os.path.join('/usr/share/sonic/device/', platform, hwsku, 'buffers.json.j2')
-    if os.path.isfile(buffer_template_file):
-        command = "{} -d -t {} >/tmp/buffers.json".format(SONIC_CFGGEN_PATH, buffer_template_file)
-        run_command(command, display_cmd=True)
-
-        qos_template_file = os.path.join('/usr/share/sonic/device/', platform, hwsku, 'qos.json.j2')
-        sonic_version_file = os.path.join('/etc/sonic/', 'sonic_version.yml')
-        if os.path.isfile(qos_template_file):
-            command = "{} -d -t {} -y {} >/tmp/qos.json".format(SONIC_CFGGEN_PATH, qos_template_file, sonic_version_file)
-            run_command(command, display_cmd=True)
+    namespace_list = [DEFAULT_NAMESPACE]
+    if sonic_device_util.get_num_npus() > 1:
+        namespace_list = sonic_device_util.get_namespaces()
+
+    if namespace:
+        if namespace not in namespace_list:
+            click.secho(
+                "Command 'qos reload' failed with invalid namespace '{}'".format(
+                    namespace
+                ),
+                fg='yellow'
+            )
+            raise click.Abort()
+        namespace_list = [namespace]
 
-            # Apply the configurations only when both buffer and qos configuration files are presented
-            command = "{} -j /tmp/buffers.json --write-to-db".format(SONIC_CFGGEN_PATH)
-            run_command(command, display_cmd=True)
-            command = "{} -j /tmp/qos.json --write-to-db".format(SONIC_CFGGEN_PATH)
+    for ns in namespace_list:
+        if ns is DEFAULT_NAMESPACE:
+            asic_id_suffix = ""
+        else:
+            asic_id = sonic_device_util.get_npu_id_from_name(ns)
+            if asic_id is None:
+                click.secho(
+                    "Command 'qos reload' failed with invalid namespace '{}'".
+                        format(ns),
+                    fg='yellow'
+                )
+                raise click.Abort()
```

Comment:
we have  sonic_device_util.get_all_namespaces() that provide the dictionary of front and back namespace with string as namespace value. Then no need to convert from id to string. We can see if this API can be used.

Reply:
I need the ASIC ID. That API doesn't give the ASIC ID.

Reply:
why can't we have directory name as "asic0" ? i feel that is better and correlate to namespace directory created by linux.
We can change port_config.ini to also be in similar folder

@judyjoseph @SuvarnaMeenakshi what do you think here ?

Reply:
This is a temporary directory, it can be anything. I still need the ASIC ID to read, and matches that format.

/usr/share/sonic/device/0
/usr/share/sonic/device/1
...

Reply:
NPU IDs and ASIC names are different. The configuration is for each NPU ID rather than namespace. I think it's OK to specify NPU ID for configuration.

Reply:
A simple approach would be to get the num of asics present in the platform and do a for range loop. We are using the asic_index as asic_id's internally.
If we derive the asic_id from namespace name, could cause problems if we create a namespace for a different purpose in the switch ?

Reply:
I'm using 'get_namespaces()' API, this should return the valid namespaces for ASICs. It's better to use common APIs so there's one place that calculates the ASIC ID and namespace. Using a range loop can be fragile if the ASIC ID allocation scheme changes.

Reply:
> This is a temporary directory, it can be anything. I still need the ASIC ID to read, and matches that format.
> 
> /usr/share/sonic/device/0
> /usr/share/sonic/device/1
> ...

@smaheshm why can't asic_id_suffix be namespace name directly ?

Reply:
because the directories use 'asic ID'.

/usr/share/sonic/device/0
/usr/share/sonic/device/1
...